dbms/src/Storages/Page/V3/spacemap/SpaceMapSTDMap.h

```diff
@@ -416,30 +331,49 @@ class STDMapSpaceMap
 
         if (it->first + it->second == it_next->first)
         {
+            deleteFromInvertIndex(it_next->second, it_next->first);
+            deleteFromInvertIndex(it->second, it->first);
+            insertIntoInvertIndex(it_next->first + it_next->second - it->first, it->first);
             free_map[it->first] = it_next->first + it_next->second - it->first;
             free_map.erase(it_next);
         }
         // next can't merge
         return true;
     }
 
+private:
+    inline void deleteFromInvertIndex(UInt64 length, UInt64 offset)
+    {
+        auto & offsets = free_map_invert_index[length];
```

Comment:
How about doing some checking here?  Like print some message if `free_map_invert_index[length]` does not exist.

Reply:
Done.